MCOL-5019: distributing cskeys secrets file, move cskeys and cspasswd functions to mcs cli.

mariadb-AlanMologorsky opened the pull request:
- [x] [add] distribute .secrets file to all nodes while adding a new node
- [x] [add] encrypt_password, generate_secrets_data, save_secrets to CEJPasswordHandler
- [x] [add] tools section to mcs cli tool
- [x] [add] mcs_cluster_tool/tools_commands.py file with cskeys and cspasswd commands
- [x] [add] cskeys and cspasswd commands to tools section of mcs cli
- [x] [mv] backup/restore commands to tools section mcs cli
- [x] [fix] minor imports ordering
- [x] [fix] constants
- [x] [fix] CEJPasswordHandler class methods to use directory for cskeys file
- [x] [fix] CEJPasswordHandler.encrypt_password to return password in hex format
- [x] [fix] CEJPasswordHandler key_length
- [x] [fix] CEJPasswordHandler os.urandom call typo
- [x] [upd] mcs cli README.md and man page
- [x] [upd] mcs cli README_DEV.md
- [x] [fix] mcs_cluster_tool/decorators.py to handle typer.Exit exception
- [x] [add] various docstrings

mariadb-AlanMologorsky committed:
feat(cmapi): MCOL-5019: distributing cskeys secrets file, move cskeys and cspasswd functions to mcs cli.

[add] distribute .secrets file to all nodes while adding a new node
[add] encrypt_password, generate_secrets_data, save_secrets to CEJPasswordHandler
[add] tools section to mcs cli tool
[add] mcs_cluster_tool/tools_commands.py file with cskeys and cspasswd commands
[add] cskeys and cspasswd commands to tools section of mcs cli
[mv] backup/restore commands to tools section mcs cli
[fix] minor imports ordering
[fix] constants

drrtuy commented on cmapi/mcs_cluster_tool/tools_commands.py:
```diff
@@ -0,0 +1,94 @@
+
+
+@handle_output
+def cskeys(
```
mcs_keys

drrtuy reviewed (approved):
untested ACK
How did you check that new passwords are compatible with the decoding mechanism used by the engine? We need a test that decodes password produced by the code added with this patch.

mariadb-AlanMologorsky committed:
feat(cmapi): MCOL-5019: review fixes.

[fix] CEJPasswordHandler class methods to use directory for cskeys file
[fix] CEJPasswordHandler.encrypt_password to return password in hex format
[fix] CEJPasswordHandler key_length
[fix] CEJPasswordHandler os.urandom call typo
[upd] mcs cli README.md and man page
[upd] mcs cli README_DEV.md
[fix] mcs_cluster_tool/decorators.py to handle typer.Exit exception
[add] various docstrings

mariadb-AlanMologorsky commented:
@drrtuy I created `.secrets` file using `cskeys` binary and then encrypt end decrypt password using `mcs cspasswd`
It confirms that my code fully compatible with old `.secrets` file generated from engine `cskeys` (decoding was compatible since I wrote this in CMAPI BTW)
And then vise versa: created `.secrets` file using `mcs cskeys` command and then encrypt and decrypt passwords using both (`cspasswd` and `mcs cspasswd`) tools. It confirms that `.secrets` file generated by mcs cli is fully compatible with old engine binaries.

mariadb-AlanMologorsky commented:
> untested ACK How did you check that new passwords are compatible with the decoding mechanism used by the engine? We need a test that decodes password produced by the code added with this patch.

I created `.secrets` file using `cskeys` binary and then encrypt end decrypt password using `mcs cspasswd`
It confirms that my code fully compatible with old `.secrets` file generated from engine `cskeys` (decoding was compatible since I wrote this in CMAPI BTW)
And then vise versa: created `.secrets` file using `mcs cskeys` command and then encrypt and decrypt passwords using both (`cspasswd` and `mcs cspasswd`) tools. It confirms that `.secrets` file generated by mcs cli is fully compatible with old engine binaries.

mariadb-AlanMologorsky commented:
> untested ACK How did you check that new passwords are compatible with the decoding mechanism used by the engine? We need a test that decodes password produced by the code added with this patch.

Anyway I could add some test for that.

As I know engine code itself only have secrets.h include in resourcemanager.cpp

drrtuy commented on cmapi/mcs_cluster_tool/README.md:
```diff
@@ -18,6 +18,8 @@ $ mcs [OPTIONS] COMMAND [ARGS]...
 * `restore`: Restore Columnstore (and/or MariaDB) data.
 * `dbrm_restore`: Restore Columnstore DBRM data.
+* `cskeys`: Generates a random AES encryption key and init vector and writes them to disk.
+* `cspasswd`: Encrypt a Columnstore plaintext password...
```
Remove the trailing '...' plz

mariadb-AlanMologorsky replied:
Done.

drrtuy reviewed (approved):
untested ACK but address the comment plz.

mariadb-AlanMologorsky committed:
Review fixes.

drrtuy reviewed (approved):
untestedACK

drrtuy commented:
Any tests @mariadb-AlanMologorsky ?
